server/src/main/java/com/tencent/rss/server/buffer/ShuffleBuffer.java

```diff
@@ -63,8 +63,8 @@ public int append(ShufflePartitionedData data) {
       for (ShufflePartitionedBlock block : data.getBlockList()) {
         blocks.add(block);
         mSize += block.getSize();
-        size += mSize;
       }
+      size += mSize;
```

Comment:
for this bug fix, there should be a test case to verify

Reply:
OK